sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -353,6 +367,30 @@ def create_or_update(
         name, version = component._get_rest_name_version()
         rest_component_resource = component._to_rest_object()
         result = None
+        try:
+            if not component._is_anonymous:
+                client_component_hash = component.get_component_hash(keys_to_omit=["version"])
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
+                remote_component_version = self.get_component_version(
+                    name=name
+                )  # will raise error if not found, so need set client_component_hash in the above line first.
+                remote_component_hash = None
+                if remote_component_version:
+                    remote_component_hash = remote_component_version.properties.properties.get("client_component_hash")
+                if remote_component_hash == client_component_hash and kwargs.get("skip_if_no_change"):
+                    component.version = remote_component_version.properties.component_spec.get(
+                        "version"
+                    )  # only update the default version component instead of creating a new version component
+                    version = component.version
+                    rest_component_resource = component._to_rest_object()
+                    logger.warning(
+                        "The component is not modified compared to the default version "
+                        "and the new version component registration is skipped."
+                    )
+                rest_component_resource.properties.properties["client_component_hash"] = client_component_hash
+        except Exception as e:
```

Comment:
Can we avoid catching a board exception here?

Reply:
It is best to add catch, because get component version will throw exception.
Regardless of whether there will be bugs in the future, I want to avoid users being unable to register and obtain components properly due to this code logic.

Reply:
what about catching `HttpResponseError` for `self._get_component_version` only? broad exception may hide some problems unexpectly.

Reply:
Of course we can catching only  for `self._get_component_version`.
But I don't want any bugs in this code logic to **block user create or update components**, because any bugs with this code logic do not affect the component create or update.
In addition, we will not hide any exceptions because our test will be blocked, and we can easily detect problems if the component is created instead of updated when the component is not changed.
So if a bug occurs, it will not block the user, but it will block our test so that we can detect it in a timely.